File: Interpreter C#/Interpreter/AssignStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class AssignStmt : IStmt
        {
            string id;
            Exp exp;

            public AssignStmt(string i,Exp e)
            {
                this.id = i;
                this.exp = e;
            }

            public override string ToString()
            {
                return this.id + " = " + this.exp.ToString();
            }

            PrgState IStmt.execute(PrgState state)
            {
                MyIStack<IStmt> stk = state.getExeStack();
                MyIDictionary<string, int> symTbl = state.getSymTable();
                int val = exp.eval(symTbl);
                if (symTbl.containsKey(this.id)==1)
                {
                    symTbl.update(id, val);
                }
                else
                {
                    symTbl.add(id, val);
                }
                return state;
            }
        }
    }
}

File: MAP/Interpretor2/src/Model/AssignStmt.java
package Model;
import Model.IStmt;
public class AssignStmt implements IStmt{
	public String id;
	public Exp exp;
	public AssignStmt(String id,Exp exp)
	{
		this.id=id;
		this.exp=exp;
	}
	@Override
	public String toString(){
		return id+"="+exp.toString();
	}
	public PrgState execute(PrgState state) 
	{
		//MyIStack<IStmt>stk=state.getExeStack();
		MyIDictionary<String,Integer>symTbl=state.getSymTable();
		MyIDictionary<Integer,Integer>heap=state.getHeap();
		try{
			int val=exp.eval(symTbl,heap);
			if (symTbl.containsKey(id) )
			{
				symTbl.update(id, val);
			}
		 else 
			 {
			 	symTbl.add(id,val);
			 }
		
		}
		catch(Exception e)
		{
			System.out.println(e);
			
		}
		return state;
	
	}
	
}
